api/platform/app.go

```diff
@@ -1,7 +1,31 @@
 package platform
 
+import (
+	"github.com/davidji99/simpleresty"
+)
+
 // App represents a Heroku app.
 type App struct {
 	ID   *string `json:"id,omitempty"`
 	Name *string `json:"name,omitempty"`
 }
+
+// AddonResponse represents a generic response from the Platform Addon API.
+type AddonResponse struct {
+	ID           *string `json:"id,omitempty"`
+	Name         *string `json:"name,omitempty"`
+	App          *App    `json:"app,omitempty"`
+	AddonService *App    `json:"addon_service,omitempty"`
+}
+
+// ListAppAddons returns the maintenance window for a postgres database.
+//
+// All times are in UTC.
+func (p *Platform) ListAppAddons(appID string) ([]*AddonResponse, *simpleresty.Response, error) {
+	var result []*AddonResponse
+	urlStr := p.http.RequestURL("/apps/%s/addons", appID)
+	// Execute the request
+	response, getErr := p.http.Get(urlStr, &result, nil)
+
+	return result, response, getErr
+}
```

Comment:
Since you aren't using any variants of the Platform API for your data source, you don't need to define a method and the associated `struct` for `AddonResponse`. You can use `heroku-go` here via the `Client` struct defined in `/api/api.go` in the resource file.

Also while this provider (and the api pkg) use my fork of [`heroku-go`](github.com/davidji99/heroku-go) in order to accomodate Platform API variants, the fork is up-to-date with [upstream](github.com/heroku/heroku-go).

Reply:
Nice catch, removed this code in favor of `heroku-go`